pyvisa_py/tcpip.py

```diff
@@ -41,6 +41,203 @@
 
 @Session.register(constants.InterfaceType.tcpip, "INSTR")
 class TCPIPInstrSession(Session):
+    """A class to dispatch to VXI11 or HiSLIP based on the protocol."""
+
+    def __new__(
+        cls,
+        resource_manager_session: VISARMSession,
+        resource_name: str,
+        parsed=None,
+        open_timeout: Optional[int] = None,
+    ):
+        obj: Union[TCPIPInstrHiSLIP, TCPIPInstrVxi11]
+
+        if parsed is None:
+            parsed = rname.parse_resource_name(resource_name)
+
+        if parsed.lan_device_name.lower().startswith("hislip"):
+            obj = TCPIPInstrHiSLIP(
+                resource_manager_session, resource_name, parsed, open_timeout
+            )
+        else:
+            obj = TCPIPInstrVxi11(
+                resource_manager_session, resource_name, parsed, open_timeout
+            )
+        return obj
+
+
+@Session.register(constants.InterfaceType.tcpip, "HISLIP")
```

Comment:
This does not really makes sense since HISLIP is not a recognized VISA resource class. I would simply remove it.

Reply:
Registering a non-existent VISA resource class may not be desirable, but without the Session.register decorator we won't get the session_type attribute defined, and that causes a problem in Session.set_attribute.  It looks like fixing that would require some major refactoring.

Reply:
The problem is that it will mess up the access to the resource class VISA attribute. At this point it sounds just better to manually add `cls.session_type = (interface_type, resource_class)` by hand rather than relying on the registering mechanism. It is not perfect but it is more correct from a user perspective.